src/pymor/models/iosys.py

```diff
@@ -505,6 +575,44 @@ def to_abcde_files(self, files_basename):
         if E is not None:
             _mmwrite(Path(files_basename + '.E'), E)
 
+    def _compute(self, solution=False, output=False, solution_d_mu=False, output_d_mu=False,
```

Comment:
Any reason why you are using `_compute` instead of `_compute_solution` and `_compute_output`. You do not seem to profit from the more complicated `_compute` implementation.

Reply:
Not sure, it's been a while. I guess this was when `input` was not included in `_compute_output`. I'll work on it.

Reply:
Ah, yes. The issue with `_compute_output` is that it doesn't give the time points of the `solution` `VectorArray`. Then the D term cannot be added (when it's nonzero).

Reply:
Also, now I see that by replacing `_compute`, I get a `CacheKeyGenerationError`.

Reply:
I get the argument with the `D` term. But can you post the backtrace of the `CacheKeyGenerationError`?

Reply:
```python
Traceback (most recent call last):
  File "test_solve.py", line 10, in <module>
    Y = fom.output(input='[sin(t[0]), sin(2 * t[0])]')
  File "/.../src/pymor/models/interface.py", line 505, in output
    data = self.compute(
  File "/.../src/pymor/models/interface.py", line 364, in compute
    retval = self.cached_method_call(self._compute_solution, mu=mu, **kwargs)
  File "/.../src/pymor/core/cache.py", line 327, in cached_method_call
    return self._cached_method_call(method, False, argnames, defaults, args, kwargs)
  File "/.../src/pymor/core/cache.py", line 353, in _cached_method_call
    key = build_cache_key((method.__name__, self_id, kwargs))
  File "/.../src/pymor/core/cache.py", line 412, in build_cache_key
    obj = transform_obj(obj)
  File "/.../src/pymor/core/cache.py", line 401, in transform_obj
    return tuple(transform_obj(o) for o in obj)
  File "/.../src/pymor/core/cache.py", line 401, in <genexpr>
    return tuple(transform_obj(o) for o in obj)
  File "/.../src/pymor/core/cache.py", line 405, in transform_obj
    return tuple((transform_obj(k), transform_obj(v)) for k, v in sorted(obj.items()))
  File "/.../src/pymor/core/cache.py", line 405, in <genexpr>
    return tuple((transform_obj(k), transform_obj(v)) for k, v in sorted(obj.items()))
  File "/.../src/pymor/core/cache.py", line 399, in transform_obj
    return transform_obj(obj._raw_values)
  File "/.../src/pymor/core/cache.py", line 405, in transform_obj
    return tuple((transform_obj(k), transform_obj(v)) for k, v in sorted(obj.items()))
  File "/.../src/pymor/core/cache.py", line 405, in <genexpr>
    return tuple((transform_obj(k), transform_obj(v)) for k, v in sorted(obj.items()))
  File "/.../src/pymor/core/cache.py", line 410, in transform_obj
    raise CacheKeyGenerationError('Cannot generate cache key for provided arguments')
pymor.core.exceptions.CacheKeyGenerationError: Cannot generate cache key for provided arguments
```

Reply:
@sdrave The above comment seems to be hidden.

Reply:
Ok, I see. We can't create cache keys for time-dependent parameters (#1810).  By implementing `_compute` you implicitly disable caching.